### ISSUE TITLE
Writing changes file ownership

### ISSUE THREAD

**Issue by kirawi** [helix-editor/helix#11988]
### Summary

Writing a file may change the file's ownership when writing.

### Reproduction Steps

```
$ touch abc
$ ls -la abc
-rw-r--r-- 1 m00n m00n 0 Oct 28 23:03 abc
$ sudo helix abc
( Write the file with any content )
$ ls -la
-rw-r--r-- 1 root m00n 4 Oct 28 23:03 abc
```

### Platform

All

### Terminal Emulator

N/a

### Installation Method

release

### Helix Version

helix 24.7

**Comment by poliorcetics** [comment]
I think the issue is https://github.com/helix-editor/helix/blob/b53dafe326d51b7f64a1c52379e3d4150cd2991e/helix-view/src/document.rs#L973, where the file is created with the access rights, owner and group of the running helix process, not that of the target file

**Comment by kirawi** [comment]
I don't think so. I believe it would open with the same file permissions if it already exists, and for backup files we already copy metadata on https://github.com/helix-editor/helix/blob/b53dafe326d51b7f64a1c52379e3d4150cd2991e/helix-view/src/document.rs#L1008-L1014

**Comment by kirawi** [comment]
Actually you might have been right. This seems to be fixed by https://github.com/helix-editor/helix/pull/11374

**Comment by Silberling** [comment]
I am evaliating helix and just ran into the same issue. It took me about 30 minutes to figure out that the permissions of config have changed and the config file was unreadable for a container.
Easy to reproduce as shown above: Create file as user ABC, open as root and you successfully shot yourself in the foot.
This behavior should not be that way. Do I understand correctly that you re-create that file? I would expect an editor to not mess with the file system unless told otherwise. If I open a file and save it the inode and fs entries should not be manipulated, only the file may be rewritten (truncated). Changing metadata of a file may also trigger audit and backup systems in a different way.

**Comment by Silberling** [comment]
I am evaliating helix and just ran into the same issue. It took me about 30 minutes to figure out that the permissions of config have changed and the config file was unreadable for a container.
Easy to reproduce as shown above: Create file as user ABC, open as root and you successfully shot yourself in the foot.
This behavior should not be that way. Do I understand correctly that you re-create that file? I would expect an editor to not mess with the file system unless told otherwise. If I open a file and save it the inode and fs entries should not be manipulated, only the file may be rewritten (truncated). Changing metadata of a file may also trigger audit and backup systems in a different way.

**Comment by waspoza** [comment]
Changing file owner is still happening. It's very annoying.

**Comment by clorteau** [comment]
Just wanted to mention it's still present. If I sudo helix a file, owner changes to root. Annoying when you don't remember it does that.

**Comment by amdavidson** [comment]
Just ran into this as well, this is a deal breaker for my usage... It's a shame as I was really liking helix.

**Comment by bbkane** [comment]
Can you make a shell script that launches helix, waits till it closes, then restores file ownership? Definitely a hack, but might be good enough for you

**Comment by kirawi** [comment]
It’s been a while so I don’t entirely remember what part of my big PR fixed it, but it should be a very simple fix IIRC.

**Comment by amdavidson** [comment]
> ... part of my big PR fixed it ...

This PR? https://github.com/helix-editor/helix/pull/14614
